packages/plugin/src/node/scanners/scanner-theia.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2015-2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { injectable } from 'inversify';
+import { pluginEngine, PluginModel, PluginPackage, PluginScanner, PluginLifecycle } from '../../common/plugin-protocol';
+
+@injectable()
+export class TheiaPluginScanner implements PluginScanner {
+    private readonly _apiType: pluginEngine = 'theiaPlugin';
+
+    get apiType(): pluginEngine {
+        return this._apiType;
+    }
+
+    getModel(plugin: PluginPackage): PluginModel {
+        return {
+            name: plugin.name,
+            publisher: plugin.publisher,
+            version: plugin.version,
+            displayName: plugin.displayName,
+            description: plugin.description,
+            engine: {
+                type: this._apiType,
+                version: plugin.engines[this._apiType]
+            },
+            entryPoint: {
+                frontend: plugin.theiaPlugin!.frontend,
+                backend: plugin.theiaPlugin!.backend
+            }
+        };
+    }
+
+    getLifecycle(plugin: PluginPackage): PluginLifecycle {
+        return {
+            startMethod: 'start',
+            stopMethod: 'stop',
+            frontendModuleName: plugin.theiaPlugin!.frontendModuleName,
+
+            backendInitPath: 'backend-init-theia.js'
```

Comment:
is there a reason why it's a .js file and not an interface with inversify binding ?

like BackendInit with theia and vscode implementation ?

so we need to bind a scanner + a init provider for each plugin type ?

Reply:
There is no such reason, so, I'll revamp this part in the proposed way.

Reply:
@benoitf 
I've started implementing your proposal but then remember that this script has to be runned on plugin host (webworker or node). So code cannot be passed through json-rpc connection.

Reply:
not webworker but separate nodejs process

Reply:
ok